sdk/core/Azure.Core/src/SyncAsyncEventArgs.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+
+namespace Azure
+{
+    /// <summary>
+    /// Provides data for <see cref="Azure.Core.SyncAsyncEventHandler{T}"/>
+    /// events that can be invoked either synchronously or asynchronously.
+    /// </summary>
+    public class SyncAsyncEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Gets a value indicating whether the event handler was invoked
+        /// synchronously or asynchronously.  Please see
+        /// <see cref="Azure.Core.SyncAsyncEventHandler{T}"/> for more details.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The same <see cref="Azure.Core.SyncAsyncEventHandler{T}"/>
+        /// event can be raised from both synchronous and asynchronous code
+        /// paths depending on whether you're calling sync or async methods on
+        /// a client.  If you write an async handler but raise it from a sync
+        /// method, the handler will be doing sync-over-async and may cause
+        /// ThreadPool starvation.  See
+        /// <see href="https://docs.microsoft.com/archive/blogs/vancem/diagnosing-net-core-threadpool-starvation-with-perfview-why-my-service-is-not-saturating-all-cores-or-seems-to-stall">
+        /// Diagnosing .NET Core ThreadPool Starvation with PerfView</see> for
```

Comment:
PerfView is a tool for maniacs. Is there a better article that describes the issue?

Reply:
I haven't seen a clearer explanation than Vance's.  I'm happy to change this if anyone has other suggestions.